### ISSUE TITLE
[selectors-4] Can an element with no boxes and no child elements be in :hover state?

### ISSUE THREAD

**Issue by bzbarsky** [w3c/csswg-drafts#1141]
Consider this testcase:

    <style>
      div { display: contents }
      div:hover { color: purple }
    </style>
    <div>This is some text</div>

When hovering over the text, does it turn purple?  The spec is pretty unclear on when an element is actually "designated with a pointing device" and what that means for display:contents elements, exactly.  Note that in this case there are no descendants of any sort matching :hover, because text nodes do not match :hover themselves.

**Comment by rune-opera** [comment]
If I read the current draft[1] correctly, it says that that the deepest node in the chain of nodes that gets the :hover state must be an element, and must generate a box. There is no requirement that the ancestors need to generate boxes to get the :hover state, though. So for this case:

```
<div style="display:contents">
  <div>
    <div style="display:contents">Hover me</div>
  </div>
</div>
```

The inner display:contents does not match :hover, but the outer does.

#1135 is related.

[1] https://drafts.csswg.org/selectors/#the-hover-pseudo

**Comment by bzbarsky** [comment]
I think the current draft leaves this totally undefined because "user designates an element" is totally undefined...  I would like us to come closer to having it be defined.

I agree that in your testcase the outer display:contents should match :hover no matter how we slice this.  The question is what should happen for the inner one.

**Comment by rune-opera** [comment]
Right, I was too focused on the "hovers over a box generated by the element" part ignoring "For example".

**Comment by tabatkins** [comment]
Intention is definitely that it gets the hover styles, whether it's the direct parent of a text node or a grandparent.  Guess I need to rephrase the "if a shadow-inclusive descendant matches :hover" to instead say "is designated by a pointing device".

**Comment by svgeesus** [comment]
> Guess I need to rephrase the "if a shadow-inclusive descendant matches :hover" to instead say "is designated by a pointing device".

This has been done, see [9.1. The Pointer Hover Pseudo-class: [:hover](https://drafts.csswg.org/selectors/#hover-pseudo)](https://drafts.csswg.org/selectors/#the-hover-pseudo)
